commerce/domain/orchestrators/classes/CartCalculateSample.cls

```diff
@@ -3,65 +3,67 @@
-        // This block is executed when starting or modifying checkout.
-        if (cartStatus == CartExtension.CartStatusEnum.CHECKOUT) {
-            // Order matters. If the promotions calculator returns an error, we will finish execution early instead of
-            // calling the other calculators
-            promotions(new CartExtension.CartCalculateCalculatorRequest(cart));
+        if (runPricing) {
+            prices(calculatorRequest);
+
+            if (hasErrorsInCVO(cvoList, CartExtension.CartValidationOutputTypeEnum.PRICING)) {
+                return;
+            }
+        }
+
+        if (runPromotions) {
+            promotions(calculatorRequest);
 
             if (hasErrorsInCVO(cvoList, CartExtension.CartValidationOutputTypeEnum.PROMOTIONS)) {
                 return;
             }
+        }
 
-            inventory(new CartExtension.CartCalculateCalculatorRequest(cart));
+        if (runInventory) {
+            inventory(calculatorRequest);
 
             if (hasErrorsInCVO(cvoList, CartExtension.CartValidationOutputTypeEnum.INVENTORY)) {
                 return;
             }
+        }
+
+        if (runShipping) {
+            shipping(calculatorRequest);
+
+            if (hasErrorsInCVO(cvoList, CartExtension.CartValidationOutputTypeEnum.SHIPPING)) {
+                return;
+            }
+        }
 
-            // To successfully place an order, each Delivery Group must have an associated Delivery Method.
-            for (Integer i = 0; i < cart.getCartDeliveryGroups().size(); i++) {
-                CartExtension.CartDeliveryGroup cartDeliveryGroup = cart.getCartDeliveryGroups().get(i);
-
-                if (cartDeliveryGroup.getDeliverToAddress() != null) {
-                    // If you don't register the Commerce_Domain_Shipping_CartCalculator extension point -
-                    // this creates a free ($0) shipping option (CartDeliveryGroupMethod).
-                    // If you register custom implementation of Commerce_Domain_Shipping_CartCalculator,
-                    // please make sure it always assigns at least one shipping option (CartDeliveryGroupMethod)
-                    // to CartDeliveryGroup.
-                    shipping(new CartExtension.CartCalculateCalculatorRequest(cart));
-
-                    if (hasErrorsInCVO(cvoList, CartExtension.CartValidationOutputTypeEnum.SHIPPING)) {
-                        return;
-                    }
-
-                    // This example takes the first available shipping option (CartDeliveryGroupMethod) and uses it as a
-                    // DeliveryMethod for the given CartDeliveryGroup.
-                    if (cartDeliveryGroup.getCartDeliveryGroupMethods().size() > 0) {
-                        String firstCartDeliveryGroupId = cartDeliveryGroup.getCartDeliveryGroupMethods().get(0).getDeliveryMethodId();
-                        cartDeliveryGroup.setDeliveryMethodId(firstCartDeliveryGroupId);
```

Comment:
do we no longer need to set a setDeliveryMethod to place an order?

Reply:
We do! This happens in `postShipping`: https://sourcegraph.soma.salesforce.com/perforce.soma.salesforce.com/app/main/core@HEAD/-/blob/commerce-checkout-impl/java/src/cartsummary/CalculatorService.java?L251 . See the `addShippingProductToCart` method in `ShippingCalculationServiceImpl.java`.